TODO/constraint-layout-hell.md

```diff
@@ -1,22 +1,24 @@
 * 原文作者：[Hafiz Waleed Hussain](http://www.uwanttolearn.com/about/)
 * 译文出自：[掘金翻译计划](https://github.com/xitu/gold-miner)
-* 译者：
+* 译者：[jifaxu](https://github.com/jifaxu)
 * 校对者：
 
-WOW, we got one more day so its time to make this day awesome by learning something new :).
+# ConstraintLayout (这到底是什么)
 
-Hello Guys, Today we are going to learn Constraint Layout in Android.
+喔，又是新的一天，为了不浪费这宝贵的时光，让我们来学点新知识吧 :)。
 
-**Motivation:**
+大家好，今天让我们学习 Android 里的 Constraint 布局。
 
-I want to discuss my own experience with this awesome layout. When Google announced this layout, I start learning but I faced a lot of issues. I try to get some good tutorials but always get Visual Editor, Drag & Drop images in tutorials. Which not make sense for me. After two months I changed my strategy. I analysed my self and I got answer. I am good in XML with Linear, Relative, Frame … layouts, so I should work with XML of Constraint Layout but again when I drag drop some views on Visual Editor and opened a XML, I am in more difficult situation because there are lot of new tags which I don’t know. I am frustrated but I never give up. Then I changed again my strategy. I decided I will not use Visual Editor, create my own Relative Layout then convert to Constraint Layout. Now magic happens 🙂 , it took only one whole day and now I am really comfortable with Constraint Layout.
+**动机：**
 
-After that I choose the same strategy with Linear Layout to Constraint, Frame Layout to Constraint. Today I am choosing same strategy with this post. Every body has different learning and thinking capacity. So may be there is a possibility you are not agree with my approach but those who are struggling, I am giving you surety this approach is awesome. One more good news, now I know how I can use Visual Editor without any confusion. Every thing is crystal clear. Its time to start playing with our best friend **CONSTRAINT LAYOUT**.
+我想先讨论一下我在学习这个很酷的布局时的经验。当 Google 发布这个布局后我就开始学习了，在这个过程中我遇到了很多的问题。我像找一些优秀的教程，但是结果都是一些教我在可视化编辑器里拖拽图片的东西，这些对我一点用都没有。两个月之后我改变了我的方法。通过分析我自己的特点我找到了答案。我擅长用 XML 来编写 LinearLayout，RelativeLayout，FrameLayout 等，所以我觉得我应该通过 XML 来学习 ConstraintLayout。但是当我在可视化编辑器里添加了一些组件并打开 XML 文件的时候，我再一次陷入了困境，这里面有太多我不认识的新属性了。虽然感到很沮丧但我并不打算就此放弃。再一次地，我改变了方法，这次我决定放弃可视化编辑器，创建一个 RelativeLayout 再将它转换成 ConstraintLayout。一切尽在意料之中，这次我只用了一天就掌握了它 🙂，现在我已经习惯使用 ConstraintLayout 了。
 
-We need to download 2.3 Android studio. In previous versions Visual Editor is not good and that show some wrong info on Design Tab. So that is really important download 2.3 beta which is available when I am writing this post.
+在这之后，我又同样的方法将 LinearLayout 和 FrameLayout 转成了 ConstraintLayout。今天我将会在这篇博客中使用同样的方法。每个人脑回路都不太一样，所以有可能你并不认同我的方法。但是对于哪些苦于不知如何入手的朋友们，我可以向你保证这个方法时值得一试的。还有一个好消息那就是现在我知道如何顺畅的使用可视化编辑器了。事情已经说的很清楚了，现在是时候开始学习 **CONSTRAINT LAYOUT** 了。
 
-Create a new project.
+首先我们需要下载 Android Studio 2.3。在这之前的可视化编辑器做的不够好，而且在 Design 标签栏里还有一些问题。所以一定要下载 2.3 beta 版。
```

Comment:
> So that is really important download 2.3 beta which is available when I am writing this post.所以一定要下载 2.3 beta 版。

此处我个人感觉是不是作者有这么个意思是建议下载2.3之后的（最新）版本，2.3beta版只是作者写这篇文章时的最新版，并非作者要求大家去下载的版本。